File: src/kotlin/FibonacciIterative.kt
// Function to calculate Fibonacci numbers iteratively
fun fibonacciIterative(n: Int): Long {
    if (n <= 1) {
        return n.toLong()
    }

    var a = 0L
    var b = 1L
    var result = 0L

    for (i in 2..n) {
        result = a + b
        a = b
        b = result
    }

    return result
}

fun main() {
    val index = 15
    println("Fibonacci (iterative) of $index is: ${fibonacciIterative(index)}")
}

File: src/go/fibonacci_iterative.go
package main

import "fmt"

func Fibonacci(n int) int {
	if n <= 1 {
		return 1
	}

	a, b := 1, 1
	for i := 2; i <= n; i++ {
		a, b = b, a+b
	}
	return b
}

func main() {
	n := 9
	fmt.Println("Fibonacci Iterative:", Fibonacci(n))
}
